crates/bevy_state/src/app.rs

```diff
@@ -104,6 +104,9 @@ impl AppExtStates for SubApp {
                 exited: None,
                 entered: Some(state),
             });
+            if S::scoped_entities_enabled() {
+                self.enable_state_scoped_entities::<S>();
```

Comment:
The docs for `enable_state_scoped_entities` should probably be updated to mention that it doesn't need to be called manually if using the attribute.